sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -162,6 +162,16 @@ case class DataSourceResolution(
 
     case DataSourceV2Relation(CatalogTableAsV2(catalogTable), _, _) =>
       UnresolvedCatalogRelation(catalogTable)
+
+    // If the database name is not provided, fallback to v1 catalog.
+    case ShowTablesStatement(None, pattern) =>
+      ShowTablesCommand(None, pattern)
```

Comment:
@rdblue Does it makes sense to fallback to v1 catalog if there is no database info passed in?

Reply:
We will fallback to v1 if no db info is provided.

Reply:
This should use the default v2 catalog if it is set.

Reply:
@rdblue currently, v2 catalog doesn't have `getCurrentDatabase` to use when database name is not provided.
What I can do is to throw an exception if default v2 catalog is set until v2 catalog supports `getCurrentDatabase`. What do you think?

Reply:
I don't think we want catalogs to support `getCurrentDatabase`. I think the current database should be tracked by Spark and passed in. @cloud-fan is working on storing this in SessionState, I think.

Reply:
@rdblue should I wait for @cloud-fan's changes go in first, or revisit this later?

Reply:
Let's default it to a constant for now and we can update it when the changes are in. If that goes in first, then we can update this PR, otherwise do it in a follow-up.

Reply:
Since you mentioned `This works for now` for throwing an exception for the case where db is not specified but default v2 catalog is set, I will leave it as it is. I will do a follow-up when the changes are in.